pkg/collector/tcp.go

```diff
@@ -77,18 +90,18 @@ func (cp *CollectingProcess) handleTCPClient(conn net.Conn) {
 			}
 			if err != nil {
 				klog.ErrorS(err, "Error when retrieving message length")
-				cp.deleteClient(address)
 				return
 			}
 			buff := make([]byte, length)
 			_, err = io.ReadFull(reader, buff)
 			if err != nil {
 				klog.ErrorS(err, "Error when reading the message")
-				cp.deleteClient(address)
 				return
 			}
 			message, err := cp.decodePacket(bytes.NewBuffer(buff), address)
 			if err != nil {
+				// TODO: should we close the connection instead and force the client to
```

Comment:
What's the pros for letting the client to re-open the connection?

Reply:
Let the client know that an invalid IPFIX message was sent and force it to re-open the connection
But I didn't make the change in this PR